### PR TITLE
add sle-es7.4 packer img

### PR DESCRIPTION
**Pull request by MalloZup**
## what does this pr?

we want to use sles-expanded support instead of centos7 for CI tests. 

This pr is the first milestone for achiving this.

We will have an image that works already with sumaform and avahi setup.

Where we will put this image? Schnell.suse.de or elsewehere?

i already tested the image with sumaform and everything works (avahi, ssh, salt).

I didn't test to swtich centos7 to sles-es but is out of scope for this

### need discussion:

where to place the img
- schnell.suse.de
- or like centos7 location

### how can you test this image:

download the image from here (tomorrow)
```m226.mgr.suse.de:/home/jenkins/workspace/sles-es7``` now i have 20% updted

### CODE REVIEW

**Comment by MalloZup** [comment]
moving image to m226.mgr.suse.de:/home/jenkins/workspace

**Comment by MalloZup** [comment]
@mcalmer  i think there were the last available on the link on schnell.suse.de. i could use the latest one if there is a link no pb. I will double check

File: packer/scripts/base-es.sh [hunk 1]
```diff
@@ -0,0 +1,11 @@
+rpm --import http://schnell.suse.de/SLES-ES/7.3/DVD1/RPM-GPG-KEY-SUSE-CLIENT-TOOLS
+rpm --import http://schnell.suse.de/SLES-ES/7.3/DVD1/RPM-GPG-KEY-Expanded-Support-20140623.asc
```
Comment (moio) [inline packer/scripts/base-es.sh]: This host is not accessible from the corporate VPN, only the R&D one. Would it be possible to retrieve the files from another host which is also accessible by colleagues in sales/support/consulting and so on?

Comment (mcalmer) [reply packer/scripts/base-es.sh]: Import a .asc file. AFAIK .asc is the signature but you want to import the key.

As long as you are in the internal network you may have access to nu.novell.com. At least for the client tools the key is available via:

http://nu.novell.com/repo/$RCE/RES7-SUSE-Manager-Tools/x86_64/repodata/repomd.xml.key

The expanded support key is not available via this way.

**Comment by mcalmer** [comment]
@MalloZup - seems there are multiple directories on `schnell.suse.de` which provide -ES images and you found the wrong one :-)

Please have a look at:
```
http://schnell.suse.de/RHEL/SLES-ES/
```

a 7.5 is planned for end of May. Zuzanna will contact infra to consolidate the dirs.

File: packer/scripts/base-es.sh (hunk 1 above)
Comment (MalloZup) [reply packer/scripts/base-es.sh]: do we need the 2nd key? maybe not, i will try to import only one

File: packer/http/ks_sles-es7.cfg [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+rootpw linux
+
+firewall --disabled
+authconfig --enableshadow --passalgo=sha512 --enablefingerprint
+selinux --disabled
+timezone --utc Europe/Berlin
+bootloader --location=mbr
+
+# repo to install salt-minion
+repo --name=SLES-ES-7.3-DVD1 --install --baseurl=http://schnell.suse.de/SLES-ES/7.3/DVD1/
```
Comment (MalloZup) [inline packer/http/ks_sles-es7.cfg]: @mcalmer  do we have any equivalent for the "http://schnell.suse.de/RHEL/SLES-ES/SLES-ES-7.4-x86_64-DVD.iso ?

Comment (mcalmer) [reply packer/http/ks_sles-es7.cfg]: extracted? No, but you can ask Rudi if he can extract it.

**Comment by MalloZup** [comment]
image rebuilded with 74 and build was fine. ( in parallel with my tests from sle15) i will upload it on /m226/workspace as sle-es74 uploading it now.

**Comment by moio** [comment]
> m226.mgr.suse.de

Can you ask Rudi or the infra team if we can get some Web space which is accessible outside of the R&D network, or if it is possible for us to just redistribute that image in the wider Internet?

**Comment by MalloZup** [comment]
@moio do you have any update for this?

**Comment by moio** [comment]
I'm still in discussions with infra people. Does this have a deadline?

**Comment by MalloZup** [comment]
from my pov this should done asap. ( there is still the cucumber integration plus small fixes that need to be done). I don#t want to procastinate until infra.

How about with put the image temporally on moio for working on the cucumber fixes needed and then move it to infra place?
the infra discussion from my pov is not something that should block this pr and other work. it can be done at anytime later and without any user diff

File: packer/http/ks_sles-es7.cfg [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+rootpw linux
+
+firewall --disabled
+authconfig --enableshadow --passalgo=sha512 --enablefingerprint
+selinux --disabled
+timezone --utc Europe/Berlin
+bootloader --location=mbr
+
+# repo to install salt-minion
+repo --name=sumatools --install --baseurl=http://download.suse.de/ibs/Devel:/Galaxy:/Manager:/Head:/RES7-SUSE-Manager-Tools/SUSE_RES-7_Update_standard/
```
Comment (moio) [inline packer/http/ks_sles-es7.cfg]: Is there a need for this repo? Can it be removed?

File: packer/sles-es7.json [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    "override": {
+      "sles-es7": {
+        "execute_command": "sh '{{.Path}}'"
+      }
+    }
+  }],
+  "builders": [{
+      "name": "sles-es7",
+      "type": "qemu",
+      "iso_url": "http://schnell.suse.de/RHEL/SLES-ES/SLES-ES-7.4-x86_64-DVD.iso",
```
Comment (moio) [inline packer/sles-es7.json]: Please use `schnell.nue.suse.com` which is also accessible from people outside the VPN

**Comment by MalloZup** [comment]
ok thx, i will do the fix asap.

Thx for understanding :+1:

File: packer/http/ks_sles-es7.cfg (hunk 3 above)
Comment (MalloZup) [reply packer/http/ks_sles-es7.cfg]: ok. 
this repo give salt 2018 when in the older one we have 2015.

We might want to use latest pkgs for the sles-es but i think we could move the update during other stage of sumaform? how we handle this for sles clients?

**Comment by MalloZup** [comment]
ok done rebuilding img

File: packer/http/ks_sles-es7.cfg (hunk 3 above)
Comment (moio) [reply packer/http/ks_sles-es7.cfg]: https://github.com/moio/sumaform/blob/master/DESIGN.md#requirements-for-base-disk-images

In base images we want the oldest possible packages, including Salt. If the oldest released Salt for RES 7 is 2015, it's totally fine.

Sumaform takes care to update any package (including Salt) at deployment time.

Comment (MalloZup) [reply packer/http/ks_sles-es7.cfg]: ok thx

**Comment by MalloZup** [comment]
build was ok
```
Build 'sles-es7' finished.

==> Builds finished. The artifacts of successful builds are:
--> sles-es7: VM files in directory: output-sles-es7
```

**Comment by moio** [comment]
Thanks! Waiting for a follow-up PR to enable this in libvirt and OpenStack :+1:

**Comment by MalloZup** [comment]
https://github.com/moio/sumaform/issues/346 openstack reminder

**Comment by admd** [comment]
@MalloZup  jfyi, you forgot to remove  "iso_checksum_type": "none".

**Pull request merged**